utils/tmux_utils.sh

```diff
@@ -54,3 +54,75 @@ setw() {
   local value=$2
   tmux_commands+=(set-window-option -gq "$option" "$value" ";")
 }
+
+get_tmux_batch_option() {
+	local option
+	option="$1"
+
+	for option_index in "${!tmux_batch_options[@]}"; do
+		IFS=":" read -r read_option read_value <<<"${tmux_batch_options[$option_index]}"
+		if [[ "$read_option" == "$option" ]]; then
+			echo "$read_value"
+			return
+		fi
+	done
+}
+
+get_interpolated_tmux_batch_option() {
+	local option
+	option="$1"
+
+	for option_index in "${!tmux_batch_options[@]}"; do
+		IFS=":" read -r read_option read_value <<<"${tmux_batch_options[$option_index]}"
+		if [[ "$read_option" == "$option" ]]; then
+			do_color_interpolation "$read_value"
+			return
+		fi
+	done
+}
+
+set_tmux_batch_option_default() {
+	local option value
+	option="$1"
+	value="$2"
+
+	tmux_batch_options+=("$option:$value")
+	tmux_batch_options_commands+=("show-option -gq $option ;")
+}
+
+set_tmux_batch_option() {
+	local option value
+	option="$1"
+	value="$2"
+
+	for option_index in "${!tmux_batch_options[@]}"; do
+		read -d ':' -r read_option <<<"${tmux_batch_options[$option_index]}"
+		if [[ "$read_option" == "$option" ]]; then
+			tmux_batch_options["$option_index"]="$option:$value"
+      # shellcheck disable=SC2034
+			tmux_batch_commands["$option_index"]="show-option -gq $option ;"
+			return
+		fi
+	done
+
+	set_tmux_batch_option_default "$option" "$value"
+}
```

Comment:
Don't keep track of default's in the array.
This will reduce the array size to not defaults but force `get_tmux_options` to walk the entire list before falling back to the default.

in `./cattpuccin.tmux` we currently have 34 `@catppuccin_...` options of which most will not be set so total compares should go down? This might be worth exploring.
I expect current implementation to be _worst_ case.

Reply:
Some quick math:
```python
>>> total_options=34
>>> compare_count_current=sum(list(range(1,total_options+1)))
>>> compare_count_current
595
>>> options_count_break_even=compare_count_current/total_options
>>> options_count_break_even
17.5
>>> 17*total_options
578
```

in formulas:

n = # options
a = # not default options
x = # compares with default in [] = n + (n-1) + ... + 2 + 1 = (n*(n+1))/2
y = worst case # compares without default in [] = a * n

break even for n => a(n):
  x(n) = y(a, n) <=> (n*(n+1))/2 = a*n <=> (n+1)/2 = a
  a(n) = (n+1)/2

  a(34) = (34+1)/2 = 35/2 = 17.5

In other words if _50% or less_ of the options are _not default_ it should be faster to not track default in the list/array